src/main/scala/leon/termination/StructuralSize.scala

```diff
@@ -66,10 +88,30 @@ trait StructuralSize {
       case TupleType(argTypes) => argTypes.zipWithIndex.map({
         case (_, index) => size(tupleSelect(expr, index + 1, true))
       }).foldLeft[Expr](InfiniteIntegerLiteral(0))(Plus)
+      case IntegerType =>
+        FunctionInvocation(typedAbsBigIntFun, Seq(expr)) 
       case _ => InfiniteIntegerLiteral(0)
     }
   }
 
+  def lexicographicDecreasing(s1: Seq[Expr], s2: Seq[Expr], strict: Boolean, sizeOfOneExpr: Expr => Expr): Expr = {
+    // Note: The Equal and GreaterThan ASTs work for both BigInt and Bitvector
+    
+    val sameSizeExprs = for ((arg1, arg2) <- (s1 zip s2)) yield Equals(sizeOfOneExpr(arg1), sizeOfOneExpr(arg2))
+    
+    val greaterBecauseGreaterAtFirstDifferentPos =
+      orJoin(for (firstDifferent <- 0 until scala.math.min(s1.length, s2.length)) yield and(
+          andJoin(sameSizeExprs.take(firstDifferent)),
+          GreaterThan(sizeOfOneExpr(s1(firstDifferent)), sizeOfOneExpr(s2(firstDifferent)))
+      ))
+      
+    if (s1.length > s2.length || (s1.length == s2.length && !strict)) {
+      or(andJoin(sameSizeExprs), greaterBecauseGreaterAtFirstDifferentPos)
```

Comment:
The reason I included this `s1.length > s2.length` is that the strict decreasing order and the weak decreasing order should be *compatible* in the sense that `forall x y,  x > y  <==>  ! y >= x`. Or in other words, `forall x y,  x > y  XOR  y >= x`.

Now for `x = (a, b, c, plus, some, stuff)` and `y = (a, b, c)`, `y >= x` does not hold, so `x > y` should hold, but that's only the case if we have this `s1.length > s2.length` condition.

Note, though, that having *compatible* `>` and `>=` is not absolutely necessary for the termination checker to work correctly. If we removed the `s1.length > s2.length` condition, the `>` relation would become a slightly smaller set, but I think all the examples would still work.

Nevertheless, I'd like to keep the `s1.length > s2.length` condition, because if you claim to implement a strict and weak lexicographic order, people will expect them to be compatible, and if later this is used somewhere else where it's required that they're compatible but they're not, it might lead to bad surprises.